yt_dlp/extractor/instagram.py

```diff
@@ -356,113 +357,196 @@ def _extract_embed_url(webpage):
-
-        display_resources = (
-            media.get('display_resources')
-            or [{'src': media.get(key)} for key in ('display_src', 'display_url')]
-            or [{'src': self._og_search_thumbnail(webpage)}])
-        thumbnails = [{
-            'url': thumbnail['src'],
-            'width': thumbnail.get('config_width'),
-            'height': thumbnail.get('config_height'),
-        } for thumbnail in display_resources if thumbnail.get('src')]
-
         return {
-            'id': video_id,
-            'formats': formats,
-            'title': media.get('title') or 'Video by %s' % username,
-            'description': description,
-            'duration': float_or_none(media.get('video_duration')),
-            'timestamp': traverse_obj(media, 'taken_at_timestamp', 'date', expected_type=int_or_none),
-            'uploader_id': traverse_obj(media, ('owner', 'id')),
-            'uploader': traverse_obj(media, ('owner', 'full_name')),
-            'channel': username,
-            'like_count': self._get_count(media, 'likes', 'preview_like') or str_to_int(self._search_regex(
-                r'data-log-event="likeCountClick"[^>]*>[^\d]*([\d,\.]+)', webpage, 'like count', fatal=False)),
-            'comment_count': self._get_count(media, 'comments', 'preview_comment', 'to_comment', 'to_parent_comment'),
-            'comments': comments,
-            'thumbnails': thumbnails,
+            'id':
+            video_id,
+            'formats':
+            formats,
+            'title':
+            f'Video by {username}',
+            'description':
+            traverse_obj(info, ('caption', 'text'), expected_type=str),
+            'duration':
+            float_or_none(info.get('video_duration')),
+            'timestamp':
+            traverse_obj(info, 'taken_at', expected_type=int_or_none),
+            'uploader_id':
+            str(traverse_obj(info, ('user', 'pk'), expected_type=int)),
+            'uploader':
+            traverse_obj(info, ('user', 'full_name'), expected_type=str),
+            'channel':
+            username,
+            'like_count':
+            traverse_obj(info, 'like_count', expected_type=int_or_none),
+            'comment_count':
+            traverse_obj(info, 'comment_count', expected_type=int_or_none),
+            'thumbnails': [{
+                'url': thumbnail['url'],
+                'width': thumbnail.get('width'),
+                'height': thumbnail.get('height'),
+            } for thumbnail in info.get('image_versions2', {}).get(
+                'candidates', []) if thumbnail.get('url')],
             'http_headers': {
                 'Referer': 'https://www.instagram.com/',
             }
         }
 
+    def _real_extract(self, url):
+        video_id, url = self._match_valid_url(url).group('id', 'url')
+        webpage, urlh = self._download_webpage_handle(url, video_id)
+        if not self._cookies_passed:
```

Comment:
that is quite literally what i just said, `--cookies`/`--cookies-from-browser` will work but `--username`/`--password` won't
```
>yt-dlp -F "https://www.instagram.com/p/CeCP_a3lT4i/" --cookies-from-browser edge
[Cookies] Extracting cookies from edge
[Cookies] Extracted 859 cookies from edge
[Instagram] CeCP_a3lT4i: Downloading webpage
[Instagram] CeCP_a3lT4i: Downloading JSON metadata
[info] Available formats for CeCP_a3lT4i:
ID                      EXT  RESOLUTION │   FILESIZE   TBR PROTO │ VCODEC                         VBR ACODEC     ABR     ASR MORE INFO
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
dash-1603632390006315ad m4a  audio only │ ~ 51.84KiB   82k https │ audio only                         mp4a.40.5  82k 44100Hz DASH audio, m4a_dash
dash-388705013189198v   webm 246x438    │ ~ 78.78KiB  125k https │ vp09.00.20.08                 125k video only             DASH video, webm_dash
dash-2183913775110963v  webm 462x822    │ ~174.12KiB  277k https │ vp09.00.30.08                 277k video only             DASH video, webm_dash
603302127410254v-0      mp4  480x854    │                  https │ vp09.00.40.08.00.01.01.01.01       unknown
603302127410254v-1      mp4  480x854    │                  https │ vp09.00.40.08.00.01.01.01.01       unknown
dash-420264599771666v   webm 720x1280   │ ~304.97KiB  486k https │ vp09.00.31.08                 486k video only             DASH video, webm_dash
dash-523880259449260v   webm 720x1280   │ ~530.49KiB  846k https │ vp09.00.31.08                 846k video only             DASH video, webm_dash
dash-574853990822999v   webm 720x1280   │ ~821.32KiB 1309k https │ vp09.00.31.08                1309k video only             DASH video, webm_dash
dash-338895704860945v   webm 720x1280   │ ~  1.19MiB 1944k https │ vp09.00.31.08                1944k video only             DASH video, webm_dash
729753078346176v        mp4  720x1280   │                  https │ vp09.00.40.08.00.01.01.01.01       unknown
dash-787399665512334vd  webm 1080x1920  │ ~  1.65MiB 2690k https │ vp09.00.40.08                2690k video only             DASH video, webm_dash

>yt-dlp -F "https://www.instagram.com/p/CeCP_a3lT4i/" --username XXX --password XXX
[Instagram] Downloading login webpage
[Instagram] Logging in
[Instagram] CeCP_a3lT4i: Downloading webpage
WARNING: [Instagram] unable to extract username; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
ERROR: [Instagram] CeCP_a3lT4i: Unable to extract video url; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

Reply:
this will also break if logged out when passing cookies, which is a problem for ppl using configs
```
>yt-dlp -F "https://www.instagram.com/p/CeCP_a3lT4i/" --cookies-from-browser firefox
[Cookies] Extracting cookies from firefox
[Cookies] Extracted 487 cookies from firefox
[Instagram] CeCP_a3lT4i: Downloading webpage
[Instagram] CeCP_a3lT4i: Downloading JSON metadata
ERROR: [Instagram] CeCP_a3lT4i: CeCP_a3lT4i: Failed to parse JSON (caused by JSONDecodeError('Expecting value: line 1 column 1 (char 0)')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

Reply:
ok i will look into this matter. If you know a fix then please share

Reply:
check if the new code works i have checked myself but didn't try out the last issue.